shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -111,7 +112,7 @@ instance FromCBOR AssetName where
   fromCBOR = do
     an <- fromCBOR
     if BS.length an > 32
-      then cborError $ DecoderErrorCustom "asset name exceeds 32 bytes:" (decodeUtf8 an)
+      then cborError $ DecoderErrorCustom "asset name exceeds 32 bytes:" (fromString $ (show . BS16.encode) an)
```

Comment:
maybe
```
decodeUtf8 . BS16.encode $ an
```
would make more sense? was there an issue?

Reply:
yeah - the error handling would not work well because asset name can be any bytestring, not necessarily a UTF8 one, so we would get an irrelevant and unexpected utf8 error (talked about this with Alex)